OsmAnd/src/net/osmand/plus/plugins/monitoring/VehicleMetricsRecordingFragment.java

```diff
@@ -0,0 +1,344 @@
+import net.osmand.plus.helpers.AndroidUiHelper;
+import net.osmand.plus.settings.backend.preferences.ListStringPreference;
+import net.osmand.plus.utils.AndroidUtils;
+import net.osmand.plus.utils.ColorUtilities;
+import net.osmand.plus.views.mapwidgets.configure.dialogs.DistanceByTapFragment;
+import net.osmand.plus.widgets.dialogbutton.DialogButton;
+import net.osmand.shared.obd.OBDCommand;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class VehicleMetricsRecordingFragment extends BaseOsmAndFragment {
+
+	public static final String TAG = DistanceByTapFragment.class.getSimpleName();
+
+	private Toolbar toolbar;
+	private ImageView navigationIcon;
+
+	private RecyclerView recyclerView;
+	private VehicleMetricsRecordingAdapter adapter;
+
+	private final Set<String> selectedCommands = new HashSet<>();
+	private final List<Object> items = new ArrayList<>();
+	private DialogButton selectAllButton;
+	private boolean allCommandsSelected = false;
+
+	@Override
+	public void onCreate(@Nullable Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+	}
+
+	@Nullable
+	@Override
+	public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+	                         @Nullable Bundle savedInstanceState) {
+		updateNightMode();
+		View view = themedInflater.inflate(R.layout.vehicle_metrics_recording_fragment, container, false);
+		AndroidUtils.addStatusBarPadding21v(requireMyActivity(), view);
+
+		toolbar = view.findViewById(R.id.toolbar);
+		AppBarLayout appBarLayout = view.findViewById(R.id.app_bar);
+		ViewCompat.setElevation(appBarLayout, 5.0f);
+
+		navigationIcon = toolbar.findViewById(R.id.close_button);
+		recyclerView = view.findViewById(R.id.recycler_view);
+		selectAllButton = view.findViewById(R.id.select_all);
+
+		ListStringPreference preference = settings.TRIP_RECORDING_VEHICLE_METRICS;
+		List<String> commands = preference.getStringsList();
+		if (commands != null) {
+			selectedCommands.addAll(preference.getStringsList());
+		}
+
+		updateAllCommandsSelected();
+		setupButtons(view);
+		setupToolbar();
+		setupItems();
+
+		return view;
+	}
+
+	@SuppressLint("NotifyDataSetChanged")
+	private void setupButtons(@NonNull View view) {
+		selectAllButton.setOnClickListener(v -> {
+			if (allCommandsSelected) {
+				selectedCommands.clear();
+			} else {
+				for (OBDCommand command : VehicleMetricsRecordingCategory.getFlatListAvailableCommands()) {
+					selectedCommands.add(command.name());
+				}
+			}
+			updateAllCommandsSelected();
+			adapter.notifyDataSetChanged();
+		});
+
+		DialogButton applyButton = view.findViewById(R.id.apply_button);
+		applyButton.setOnClickListener(v -> {
+			settings.TRIP_RECORDING_VEHICLE_METRICS.setStringsList(new ArrayList<>(selectedCommands));
+			Fragment fragment = getTargetFragment();
```

Comment:
If there are no changes, do we need to make the apply button disabled? something similar to [this](https://github.com/osmandapp/Osmand/blob/ddc5ceb39b7f785afd9fa02f9bef3a61a12dd58c/OsmAnd/src/net/osmand/plus/quickaction/MapButtonAppearanceFragment.java#L213)?